promql/engine.go

```diff
@@ -207,13 +207,28 @@ func contextErr(err error, env string) error {
 	}
 }
 
+// QueryTracker should provide access to two features: tracking of active query, that are logged on restart,
+// and control over maximum number of concurrent requests.
```

Comment:
> control over maximum number of concurrent requests

From the looks of it, the interface does not give you control over the max query and rather just tells you what is the max possible concurrent queries. So is it something missing in the interface or the comments needs an update?

If max queries is a requirement, then the description of `Insert` must say that `This call must block...` instead of `can block...`.

Reply:
> From the looks of it, the interface does not give you control over the max query and rather just tells you what is the max possible concurrent queries. So is it something missing in the interface or the comments needs an update?

I see what you mean. I can update the comment.

> If max queries is a requirement, then the description of `Insert` must say that `This call must block...` instead of `can block...`.

My view is that it's really up to the implementation to decide whether to do blocking or not, caller just needs to be aware of that possibility and work with it.

Reply:
I've updated the comments, please take a look.